tasks/delete_old_reconcile_jobs/test/unit_tests/test_delete_old_reconcile_jobs.py

```diff
@@ -0,0 +1,313 @@
+            [
+                {"job_ids_to_delete": ids}
+            ],
+        )
+
+        mock_exit.assert_called_once_with(None, None, None)
+
+    @patch("delete_old_reconcile_jobs.LOGGER")
+    @patch("delete_old_reconcile_jobs.delete_jobs_sql")
+    def test_get_jobs_older_than_x_days_error_logged_and_raised(
+            self,
+            mock_delete_jobs_sql: MagicMock,
+            mock_logger: MagicMock,
+    ):
+        """
+        Exceptions from Postgres should bubble up.
+        """
+        ids = [random.randint(0, 9999), random.randint(0, 9999)]
+        expected_exception = Exception(uuid.uuid4().__str__())
+        mock_execute = Mock(side_effect=expected_exception)
+        mock_connection = Mock()
+        mock_connection.execute = mock_execute
+        mock_exit = Mock(return_value=False)
+        mock_enter = Mock()
+        mock_enter.__enter__ = Mock(return_value=mock_connection)
+        mock_enter.__exit__ = mock_exit
+        mock_engine = Mock()
+        mock_engine.begin = Mock(return_value=mock_enter)
+
+        with self.assertRaises(Exception) as cm:
+            delete_old_reconcile_jobs.delete_jobs(
+                ids,
+                mock_engine
+            )
+        self.assertEqual(expected_exception, cm.exception)
+        mock_enter.__enter__.assert_called_once_with()
+
+        mock_exit.assert_called_once_with(
+            Exception, expected_exception, unittest.mock.ANY
+        )
+        mock_logger.error.assert_called_once_with(
+            f"Error while deleting jobs: {expected_exception}"
+        )
+
+    # noinspection PyPep8Naming
+    @patch("orca_shared.database.shared_db.get_configuration")
+    @patch("delete_old_reconcile_jobs.LOGGER")
+    @patch("delete_old_reconcile_jobs.task")
+    @patch.dict(
+        os.environ,
+        {
+            "DB_CONNECT_INFO_SECRET_ARN": "test"
+        },
+        clear=True,
+     )
+    def test_handler_happy_path(
+        self,
+        mock_task: MagicMock,
+        mock_LOGGER: MagicMock,
+        mock_get_configuration: MagicMock,
+    ):
+        """
+        Happy path for handler assembling information to call Task.
+        """
+        internal_reconciliation_expiration_days = random.randint(1, 300)
+
+        mock_context = Mock()
+        event = Mock()
+
+        with patch.dict(
+            os.environ,
+            {
+                delete_old_reconcile_jobs.OS_ENVIRON_INTERNAL_RECONCILIATION_EXPIRATION_DAYS: str(internal_reconciliation_expiration_days)
+            },
+        ):
+            delete_old_reconcile_jobs.handler(event, mock_context)
+
+        mock_LOGGER.setMetadata.assert_called_once_with(event, mock_context)
+        mock_get_configuration.assert_called_once_with(os.environ["DB_CONNECT_INFO_SECRET_ARN"])
+        mock_task.assert_called_once_with(
+            internal_reconciliation_expiration_days,
+            mock_get_configuration.return_value,
+        )
+
+    # copied from shared_db.py
```

Comment:
Likely can remove lines 279 - 313 since the function is not used in the main code and could be removed.

Reply:
Removed